pkg/distributor/push.go

```diff
@@ -207,43 +210,15 @@ func calculateRetryAfter(retryAttemptHeader string, baseSeconds int, maxBackoffE
 // toHTTPStatus converts the given error into an appropriate HTTP status corresponding
 // to that error, if the error is one of the errors from this package. Otherwise, an
 // http.StatusInternalServerError is returned.
-// to that error, if the error is one of the errors from this package. Otherwise, codes.Internal and
-// http.StatusInternalServerError is returned.
 func toHTTPStatus(ctx context.Context, pushErr error, limits *validation.Overrides) int {
-	if errors.Is(pushErr, context.DeadlineExceeded) {
-		return http.StatusInternalServerError
-	}
```

Comment:
Is it wise to remove this? I guess a possibility is you would receive an `Error` wrapping a `context.DeadlineExceeded`. Why not just keep it?

Reply:
You are right. Thank you